<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.33">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Leonardo Collado Torres">
<meta name="description" content="Overview course information for BSPH Biostatistics 140.776">

<title>01 - Welcome! – Statistical Computing (BSPH 140.776)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../icon_32.png" rel="icon" type="image/png">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-34f7cc368464e0fe4b62d8d4059141d4.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Statistical Computing (BSPH 140.776)</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-general-information" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">General Information</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-general-information">    
        <li>
    <a class="dropdown-item" href="../../syllabus.html">
 <span class="dropdown-text">Syllabus</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../schedule.html">
 <span class="dropdown-text">Schedule</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-course-materials" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Course Materials</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-course-materials">    
        <li>
    <a class="dropdown-item" href="../../lectures.html">
 <span class="dropdown-text">Lectures</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../projects.html">
 <span class="dropdown-text">Projects</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../resources.html">
 <span class="dropdown-text">Resources</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/lcolladotor/jhustatcomputing"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/lcolladotor"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">01 - Welcome!</h1>
                  <div>
        <div class="description">
          Overview course information for BSPH Biostatistics 140.776
        </div>
      </div>
                          <div class="quarto-categories">
                <div class="quarto-category">course-admin</div>
                <div class="quarto-category">module 1</div>
                <div class="quarto-category">week 1</div>
              </div>
                  </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-heading">Affiliations</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author"><a href="http://lcolladotor.github.io/">Leonardo Collado Torres</a> </p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              <a href="https://libd.org/">
              Lieber Institute for Brain Development
              </a>
            </p>
          <p class="affiliation">
              <a href="https://publichealth.jhu.edu/departments/biostatistics">
              Johns Hopkins Bloomberg School of Public Health Department of Biostatistics
              </a>
            </p>
        </div>
    </div>

  <div class="quarto-title-meta">

        
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#assumptions-and-pre-requisites" id="toc-assumptions-and-pre-requisites" class="nav-link active" data-scroll-target="#assumptions-and-pre-requisites">Assumptions and pre-requisites</a></li>
  <li><a href="#getting-set-up" id="toc-getting-set-up" class="nav-link" data-scroll-target="#getting-set-up">Getting set up</a></li>
  <li><a href="#learning-objectives" id="toc-learning-objectives" class="nav-link" data-scroll-target="#learning-objectives">Learning Objectives</a></li>
  <li><a href="#course-logistics" id="toc-course-logistics" class="nav-link" data-scroll-target="#course-logistics">Course logistics</a>
  <ul class="collapse">
  <li><a href="#course-staff" id="toc-course-staff" class="nav-link" data-scroll-target="#course-staff">Course Staff</a></li>
  <li><a href="#assignment-due-dates" id="toc-assignment-due-dates" class="nav-link" data-scroll-target="#assignment-due-dates">Assignment Due Dates</a></li>
  <li><a href="#grading" id="toc-grading" class="nav-link" data-scroll-target="#grading">Grading</a></li>
  <li><a href="#submitting-assignments" id="toc-submitting-assignments" class="nav-link" data-scroll-target="#submitting-assignments">Submitting assignments</a></li>
  <li><a href="#reproducibility" id="toc-reproducibility" class="nav-link" data-scroll-target="#reproducibility">Reproducibility</a></li>
  <li><a href="#code-of-conduct" id="toc-code-of-conduct" class="nav-link" data-scroll-target="#code-of-conduct">Code of Conduct</a></li>
  <li><a href="#need-help" id="toc-need-help" class="nav-link" data-scroll-target="#need-help">Need Help?</a></li>
  <li><a href="#feedback" id="toc-feedback" class="nav-link" data-scroll-target="#feedback">Feedback</a></li>
  <li><a href="#license-and-attribution" id="toc-license-and-attribution" class="nav-link" data-scroll-target="#license-and-attribution">License and attribution</a></li>
  <li><a href="#academic-ethics" id="toc-academic-ethics" class="nav-link" data-scroll-target="#academic-ethics">Academic Ethics</a></li>
  <li><a href="#disability-support-service" id="toc-disability-support-service" class="nav-link" data-scroll-target="#disability-support-service">Disability Support Service</a></li>
  <li><a href="#previous-versions-of-the-class" id="toc-previous-versions-of-the-class" class="nav-link" data-scroll-target="#previous-versions-of-the-class">Previous versions of the class</a></li>
  <li><a href="#typos-and-corrections" id="toc-typos-and-corrections" class="nav-link" data-scroll-target="#typos-and-corrections">Typos and corrections</a></li>
  </ul></li>
  <li><a href="#r-session-information" id="toc-r-session-information" class="nav-link" data-scroll-target="#r-session-information">R session information</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<p><em>This lecture, as the rest of the course, is adapted from the version <a href="https://www.stephaniehicks.com/">Stephanie C. Hicks</a> designed and maintained in 2021 and 2022. Check the recent changes to this file through the <a href="https://github.com/lcolladotor/jhustatcomputing/commits/main/posts/01-welcome/index.qmd">GitHub history</a>.</em></p>
<p>Welcome! I am very excited to have you in our one-term (i.e.&nbsp;half a semester) course on Statistical Computing course number (140.776) offered by the <a href="https://publichealth.jhu.edu/departments/biostatistics">Department of Biostatistics</a> at the <a href="https://publichealth.jhu.edu">Johns Hopkins Bloomberg School of Public Health</a>.</p>
<iframe style="border-radius:12px" src="https://open.spotify.com/embed/track/6N4bifNtBKSSDmIL3qZRh8?utm_source=generator" width="100%" height="352" frameborder="0" allowfullscreen="" allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture" loading="lazy">
</iframe>
<blockquote class="twitter-tweet blockquote">
<p lang="en" dir="ltr">
</p><p>I’m excited to be back this year to teach “140.776 Statistical Computing” at <a href="https://twitter.com/jhubiostat?ref_src=twsrc%5Etfw"><span class="citation" data-cites="jhubiostat">@jhubiostat</span></a> <a href="https://twitter.com/JohnsHopkinsSPH?ref_src=twsrc%5Etfw"><span class="citation" data-cites="JohnsHopkinsSPH">@JohnsHopkinsSPH</span></a> 😊<br><br>This year, I decided to start with some music 🎶. I like part of the lyrics of this song, which talks about overcoming challenges, […]<a href="https://t.co/H5Uq6QhH4D">https://t.co/H5Uq6QhH4D</a><br><br>1/3 🧵 <a href="https://t.co/pkRjefcokc">pic.twitter.com/pkRjefcokc</a></p>
<p></p>
<p>— 🇲🇽 Leonardo Collado-Torres (<span class="citation" data-cites="lcolladotor">@lcolladotor</span>) <a href="https://twitter.com/lcolladotor/status/1828455658481430960?ref_src=twsrc%5Etfw">August 27, 2024</a></p>
</blockquote>
<script async="" src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
<p>This course is designed for ScM and PhD students at Johns Hopkins Bloomberg School of Public Health. I am pretty flexible about permitting outside students, but I want everyone to be aware of the goals and assumptions so no one feels like they are surprised by how the class works.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The primary goal of the course is to teach you practical programming and computational skills required for the research and application of statistical methods.</p>
</div>
</div>
<p>This class is not designed to teach the theoretical aspects of statistical or computational methods, but rather the goal is to help with the <strong>practical issues</strong> related to setting up a statistical computing environment for data analyses, developing high-quality R packages, conducting reproducible data analyses, best practices for data visualization and writing code, and creating websites for personal or project use.</p>
<section id="assumptions-and-pre-requisites" class="level3">
<h3 class="anchored" data-anchor-id="assumptions-and-pre-requisites">Assumptions and pre-requisites</h3>
<p>The course is designed for students in the Johns Hopkins Biostatistics Masters and PhD programs. However, we do not assume a significant background in statistics. Specifically we assume:</p>
<section id="you-know-the-basics-of-at-least-one-programming-language-e.g.-r-or-python" class="level4">
<h4 class="anchored" data-anchor-id="you-know-the-basics-of-at-least-one-programming-language-e.g.-r-or-python">1. You know the <strong>basics</strong> of at least one programming language (e.g.&nbsp;R or Python)</h4>
<ul>
<li>If it’s not R, we assume that you are willing to spend the time to learn R</li>
<li>You have heard of things such as control structures, functions, loops, etc</li>
<li>Know the difference between different data types (e.g.&nbsp;character, numeric, etc)</li>
<li>Know the basics of plotting (e.g.&nbsp;what is a scatterplot, histogram, etc)</li>
</ul>
</section>
<section id="you-know-the-basics-of-computing-environments" class="level4">
<h4 class="anchored" data-anchor-id="you-know-the-basics-of-computing-environments">2. You know the <strong>basics</strong> of computing environments</h4>
<ul>
<li>You have access to a computing environment (i.e.&nbsp;locally on a laptop or working in the cloud)</li>
<li>You generally feel comfortable with installing and working with software</li>
</ul>
</section>
<section id="you-know-the-basics-of-statistics" class="level4">
<h4 class="anchored" data-anchor-id="you-know-the-basics-of-statistics">3. You know the <strong>basics</strong> of statistics</h4>
<ul>
<li>The central dogma (estimates, standard errors, basic distributions, etc.)</li>
<li>Key statistical terms and methods</li>
<li>Differences between estimation vs testing vs prediction</li>
<li>Know how to fit and interpret <strong>basic</strong> statistical models (e.g.&nbsp;linear models)</li>
</ul>
</section>
<section id="you-know-the-basics-of-reproducible-research" class="level4">
<h4 class="anchored" data-anchor-id="you-know-the-basics-of-reproducible-research">4. You know the <strong>basics</strong> of reproducible research</h4>
<ul>
<li>Difference between replication and reproducible. If not, check this excellent paper by Prasad Patil et al.: <a href="https://doi.org/10.1038/s41562-019-0629-z" class="uri">https://doi.org/10.1038/s41562-019-0629-z</a>.</li>
<li>Know how to cite references (e.g.&nbsp;like in a publication)</li>
<li>Somewhat familiar with tools that enable reproducible research (In complete transparency, we will briefly cover these topics in the first week, but depending on your comfort level with them, this may impact whether you choose to continue with the course).</li>
</ul>
<p>Since the target audience for this course is advanced students in statistics we will not be able to spend significant time covering these concepts and technologies. To give you some idea about how these prerequisites will impact your experience in the course, we will be using R for nearly all classes, we will be turning in all assignments via R Markdown documents and you will be encouraged (not required) to use git/GitHub to track changes to your code over time. The majority of the assignments will involve learning the practical issues around performing data analyses, building software packages, building websites, etc all using the R programming language. Data analyses you will perform will also often involve significant data extraction, cleaning, and transformation. We will learn about tools to do all of this, but hopefully most of this sounds familiar to you so you can focus on the concepts we will be teaching around best practices for statistical computing.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>Some resources that may be useful if you feel you may be missing pieces of this background:</p>
<ul>
<li><strong>Statistics</strong> - <a href="https://www.coursera.org/learn/biostatistics">Mathematical Biostatistics Bootcamp I (Coursera)</a>; <a href="https://www.coursera.org/learn/biostatistics-2">Mathematical Biostatistics Bootcamp II (Coursera)</a></li>
<li><strong>Basic Data Science</strong> - <a href="https://leanpub.com/universities/set/jhu/cloud-based-data-science">Cloud Data Science (Leanpub)</a>; <a href="https://www.coursera.org/specializations/jhu-data-science">Data Science Specialization (Coursera)</a></li>
<li><strong>Version Control</strong> - <a href="https://lab.github.com/">Github Learning Lab</a>; <a href="https://happygitwithr.com/">Happy Git and Github for the useR</a></li>
<li><strong>Rmarkdown</strong> - <a href="https://rmarkdown.rstudio.com/lesson-1.html">Rmarkdown introduction</a></li>
<li><strong>R 101</strong> <a href="https://research.libd.org/rstatsclub/"><em>LIBD rstats club</em></a> blog post: <a href="https://research.libd.org/rstatsclub/2018/12/24/r_101/" class="uri">https://research.libd.org/rstatsclub/2018/12/24/r_101/</a></li>
<li><strong>Introductory R videos</strong> from the <a href="https://research.libd.org/rstatsclub/"><em>LIBD rstats club</em></a> such as these videos:</li>
</ul>
<iframe width="560" height="315" src="https://www.youtube.com/embed/T51DRWeuKm8" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen="">
</iframe>
<iframe width="560" height="315" src="https://www.youtube.com/embed/XTuJ8vGnzBU" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen="">
</iframe>
</div>
</div>
</section>
</section>
<section id="getting-set-up" class="level3">
<h3 class="anchored" data-anchor-id="getting-set-up">Getting set up</h3>
<p>You must install <a href="https://cran.r-project.org">R</a> and <a href="https://posit.co/products/open-source/rstudio/?sid=1">RStudio IDE</a> or <a href="https://positron.posit.co/download.html">Positron</a> (available from <a href="https://posit.co/">Posit</a>) on your computing environment in order to complete this course.</p>
<p>These are three <strong>different</strong> applications that must be installed separately before they can be used together:</p>
<ul>
<li><p><code>R</code> is the core underlying programming language and computing engine that we will be learning in this course</p></li>
<li><p><code>RStudio IDE</code> is an interface into R that makes many aspects of using and programming R simpler</p></li>
<li><p><code>Positron</code> is a new interface for working with R developed by <a href="https://posit.co/">Posit</a> (formerly RStudio). All new features are being implemented in <code>Positron</code>, but as it was just recently released, it might have some bugs. If you are choosing between <code>RStudio IDE</code> and <code>Positron</code>, check <a href="https://positron.posit.co/faqs.html#is-rstudio-going-away" class="uri">https://positron.posit.co/faqs.html#is-rstudio-going-away</a>.</p></li>
</ul>
<p><code>R</code>, <code>RStudio IDE</code>, and <code>Positron</code> are available for Windows, macOS, and most flavors of Unix and Linux. Please download the version that is suitable for your computing setup.</p>
<p>Throughout the course, we will make use of numerous R add-on packages that must be installed over the Internet. Packages can be installed using the <code>install.packages()</code> function in R. For example, to install the <code>tidyverse</code> package, you can run</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>in the R console.</p>
<p>We will also be encourage you to use <a href="https://git-scm.com/">Git</a>, following the <a href="https://happygitwithr.com/"><em>Happy Git and GitHub for the useR</em></a> book, to version control your code. Through the <a href="https://github.com/education">GitHub Education</a> pack we will get free access to <a href="https://github.com/features/copilot">GittHub Copilot Pro</a> to help us write code: think about it as an enhanced auto-complete. Finally, we’ll use <a href="https://posit-dev.github.io/air/">Air</a> for automatically formatting our R code.</p>
<section id="how-to-download-r-for-windows" class="level4">
<h4 class="anchored" data-anchor-id="how-to-download-r-for-windows">How to Download R for Windows</h4>
<p>Go to <a href="https://cran.r-project.org" class="uri">https://cran.r-project.org</a> and</p>
<ol type="1">
<li><p>Click the link to “Download R for Windows”</p></li>
<li><p>Click on “base”</p></li>
<li><p>Click on “Download R 4.5.1 for Windows”</p></li>
</ol>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>The version in the video is not the latest version of R. Please download the latest version.</p>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../../videos/downloadRWindows.gif" class="img-fluid figure-img" style="width:80.0%"></p>
<figcaption>Video Demo for Downloading R for Windows</figcaption>
</figure>
</div>
</section>
<section id="how-to-download-r-for-the-mac" class="level4">
<h4 class="anchored" data-anchor-id="how-to-download-r-for-the-mac">How to Download R for the Mac</h4>
<p>Go to <a href="https://cran.r-project.org" class="uri">https://cran.r-project.org</a> and</p>
<ol type="1">
<li><p>Click the link to “Download R for (Mac) OS X”.</p></li>
<li><p>Click on “R-4.5.1-arm64.pkg” (for Apple Silicon Macs)</p></li>
</ol>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>The version in the video is not the latest version of R. Please download the latest version.</p>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../../videos/downloadRMac.gif" class="img-fluid figure-img" style="width:80.0%"></p>
<figcaption>Video Demo for Downloading R for the Mac</figcaption>
</figure>
</div>
</section>
<section id="how-to-download-rstudio-ide" class="level4">
<h4 class="anchored" data-anchor-id="how-to-download-rstudio-ide">How to Download <code>RStudio IDE</code></h4>
<p>Go to <a href="https://posit.co/" class="uri">https://posit.co/</a> (formerly <a href="">https://rstudio.com</a>) and</p>
<ol type="1">
<li><p>Click on “Free &amp; Open Source” in the top menu</p></li>
<li><p>Click on “RStudio IDE” in the drop down menu</p></li>
<li><p>Click the button that says “DOWNLOAD RSTUDIO DESKTOP”</p></li>
<li><p>Click the button under “Download RStudio” which will take you to <a href="https://posit.co/download/rstudio-desktop/" class="uri">https://posit.co/download/rstudio-desktop/</a></p></li>
<li><p>Under the section “All Installers and Tarballs” choose the file that is appropriate for your operating system.</p></li>
</ol>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>NOTE: The video shows how to download RStudio for the Mac but you should download RStudio for whatever computing setup you have</p>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../../videos/downloadRStudio.gif" class="img-fluid figure-img" style="width:80.0%"></p>
<figcaption>Video Demo for Downloading RStudio</figcaption>
</figure>
</div>
</section>
<section id="how-to-download-positron" class="level4">
<h4 class="anchored" data-anchor-id="how-to-download-positron">How to Download Positron</h4>
<p>Go to <a href="https://posit.co/" class="uri">https://posit.co/</a> and</p>
<ol type="1">
<li><p>Click on “Free &amp; Open Source” in the top menu</p></li>
<li><p>Click on “Positron” in the drop down menu</p></li>
<li><p>Click on “Download Positron” which will take you to <a href="https://positron.posit.co/download.html" class="uri">https://positron.posit.co/download.html</a></p></li>
<li><p>Read the license agreement.</p></li>
<li><p>Accept the license agreement.</p></li>
<li><p>On the table with all the download options, choose the file that is appropriate for your operating system.</p></li>
</ol>
</section>
<section id="how-to-download-git" class="level4">
<h4 class="anchored" data-anchor-id="how-to-download-git">How to Download Git</h4>
<p>Install <code>Git</code> on your computer following the detailed instructions at <a href="https://happygitwithr.com/install-git" class="uri">https://happygitwithr.com/install-git</a>, which will depend on your operating system.</p>
</section>
<section id="create-your-github-account" class="level4">
<h4 class="anchored" data-anchor-id="create-your-github-account">Create your GitHub account</h4>
<p>Create a personal GitHub account following the instructions at <a href="https://docs.github.com/en/get-started/signing-up-for-github/signing-up-for-a-new-github-account" class="uri">https://docs.github.com/en/get-started/signing-up-for-github/signing-up-for-a-new-github-account</a>.</p>
<p>Following <a href="https://bsky.app/profile/jtleek.bsky.social">Jeff Leek</a>’s advice on <a href="https://leanpub.com/modernscientist"><strong>How to be a modern scientist</strong></a> that was part of my team’s bootcamps at <a href="https://lcolladotor.github.io/bioc_team_ds/how-to-be-a-modern-scientist.html" class="uri">https://lcolladotor.github.io/bioc_team_ds/how-to-be-a-modern-scientist.html</a>, try to choose a username that will be the same one you use on your email and other work-related social media platforms such as Bluesky, LinkedIn, etc.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/rcX-VyxHEXA" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen="">
</iframe>
<p>If you are using <code>Positron</code>, under the “Accounts” options (bottom left corner), click on “Sign in with GitHub […]”, provide the 8 digit code, and authorize “posit-dev”.</p>
</section>
<section id="how-to-apply-for-github-education-and-configure-github-copilot" class="level4">
<h4 class="anchored" data-anchor-id="how-to-apply-for-github-education-and-configure-github-copilot">How to Apply for GitHub Education and configure GitHub Copilot</h4>
<p>Go to <a href="https://github.com/education" class="uri">https://github.com/education</a> and</p>
<ol type="1">
<li>Click on “Join GitHub education” (likely <a href="https://github.com/settings/education/benefits?locale=en-US" class="uri">https://github.com/settings/education/benefits?locale=en-US</a>)</li>
<li>Sign in to your GitHub account if you haven’t logged in</li>
<li>Click on “Start Application”</li>
<li>There will be a few options to verify that you are a student. One of them is by having a photo of your student ID with dates showing that your student ID is still valid. I’ve used a screenshot of my Faculty website at <a href="https://publichealth.jhu.edu/faculty/4647/leonardo-collado-torres" class="uri">https://publichealth.jhu.edu/faculty/4647/leonardo-collado-torres</a> when validating my information.</li>
<li>Wait a few days for GitHub to process your application.</li>
<li>Once your application has been approved, if you are using <code>RStudio IDE</code> follow the instructions at <a href="https://docs.posit.co/ide/user/ide/guide/tools/copilot.html" class="uri">https://docs.posit.co/ide/user/ide/guide/tools/copilot.html</a> to enable GitHub Copilot in RStudio IDE. I have the “Index project files with GitHub Copilot” checkbox enabled on my preferences.</li>
<li>If you are using <code>Positron</code>, follow the instructions at <a href="https://positron.posit.co/assistant" class="uri">https://positron.posit.co/assistant</a> to enable “Positron assistant”. Then under the “Positron Assistant” chat, click on “Add Anthropic as a Chat Provider”, then select “GitHub Copilot” and click “Sign in”.
<ol type="1">
<li>Note that you can in theory get API access to Anthropic through <a href="https://ailabsupport.it.jh.edu/faq/" class="uri">https://ailabsupport.it.jh.edu/faq/</a> if you pay for it. Or you can pay out of pocket for it at <a href="https://claude.ai/login" class="uri">https://claude.ai/login</a>.</li>
</ol></li>
</ol>
</section>
<section id="how-to-get-air-auto-code-formatter" class="level4">
<h4 class="anchored" data-anchor-id="how-to-get-air-auto-code-formatter">How to get Air (auto-code formatter)</h4>
<p>Go to <a href="https://posit-dev.github.io/air/" class="uri">https://posit-dev.github.io/air/</a> and</p>
<ol type="1">
<li><p>Click on “RStudio” if you are using <code>RStudio IDE</code>. That will take you to <a href="https://posit-dev.github.io/air/editor-rstudio.html" class="uri">https://posit-dev.github.io/air/editor-rstudio.html</a> where you will find the detailed installation instructions.</p>
<ul>
<li>Note that I recommend enabling the <code>Tools -&gt; Global Options -&gt; Code -&gt; Saving</code> and check <code>Reformat documents on save</code> option. Particularly if you are working on your own code.</li>
</ul></li>
<li><p>Click on “VS Code and Positron” if you are using <code>Positron</code>. That will take you to <a href="https://posit-dev.github.io/air/editor-vscode.html" class="uri">https://posit-dev.github.io/air/editor-vscode.html</a> where some options and configuration steps are described. Note however, that <code>Air</code> is already available from <code>Positron</code>. You will need to open your <code>settings.json</code> file from the Command Palette (<code>Shift + Cmd + P</code> on Mac/Linux, <code>Shift + Ctrl + P</code> on Windows):</p>
<ul>
<li><p>Run <code>Preferences: Open Workspace Settings (JSON)</code> to modify workspace specific settings (recommended).</p></li>
<li><p>Run <code>Preferences: Open User Settings (JSON)</code> to modify global user settings (preferred option if you are not working with code from others). Then edit your <code>settings.json</code> with the relevant lines about R code formatting and <code>Air</code> as shown below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode json code-with-copy"><code class="sourceCode json"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">{</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"files.associations"</span><span class="fu">:</span> <span class="fu">{</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"renv.lock"</span><span class="fu">:</span> <span class="st">"json"</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">},</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"positron.assistant.gitIntegration.enable"</span><span class="fu">:</span> <span class="kw">true</span><span class="fu">,</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"positron.assistant.showTokenUsage.enable"</span><span class="fu">:</span> <span class="kw">true</span><span class="fu">,</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"positron.assistant.toolDetails.enable"</span><span class="fu">:</span> <span class="kw">true</span><span class="fu">,</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"[r]"</span><span class="fu">:</span> <span class="fu">{</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"editor.formatOnSave"</span><span class="fu">:</span> <span class="kw">true</span><span class="fu">,</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"editor.defaultFormatter"</span><span class="fu">:</span> <span class="st">"Posit.air-vscode"</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">}</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></li>
</ul>
<ol type="1">
<li>You can also configure a few options at <a href="positron://settings/air.dependencyLogLevels">positron://settings/air.dependencyLogLevels</a> and related settings.</li>
</ol></li>
</ol>
</section>
</section>
<section id="learning-objectives" class="level1">
<h1>Learning Objectives</h1>
<p>The goal is by the end of the class, students will be able to:</p>
<ol type="1">
<li>Install and configure software necessary for a statistical programming environment</li>
<li>Discuss generic programming language concepts as they are implemented in a high-level statistical language</li>
<li>Write and debug code in base R and the tidyverse (and integrate code from Python modules)</li>
<li>Build basic data visualizations using R and the tidyverse</li>
<li>Discuss best practices for coding and reproducible research, basics of data ethics, basics of working with special data types, and basics of storing data
<ul>
<li>Example of why keeping track of R versions is important: <a href="https://github.com/edward130603/BayesSpace/issues/150" class="uri">https://github.com/edward130603/BayesSpace/issues/150</a>.</li>
</ul></li>
</ol>
</section>
<section id="course-logistics" class="level1">
<h1>Course logistics</h1>
<p>The only option is to take the course is in-person (140.776.01).</p>
<ul>
<li><a href="https://lcolladotor.github.io/jhustatcomputing" class="uri">https://lcolladotor.github.io/jhustatcomputing</a></li>
</ul>
<p>All communication for the course is going to take place on one of three platforms:</p>
<ul>
<li><p><strong>Courseplus</strong>: for discussion, sharing resources, collaborating, and announcements</p>
<ul>
<li>Important for practicing written communication in the context of asking for help with R</li>
</ul></li>
<li><p><strong>Github</strong>: for getting access to course materials (e.g.&nbsp;lectures, project assignments)</p>
<ul>
<li>Course Github: <a href="https://github.com/lcolladotor/jhustatcomputing" class="uri">https://github.com/lcolladotor/jhustatcomputing</a></li>
</ul></li>
<li><p><strong>Lectures</strong>: lectures will be in person</p>
<ul>
<li>All in person lectures to be recorded and posted online after class ends</li>
<li>If for some reason I am sick or not capable of coming onsite, I will send out a Zoom link for everyone to attend remotely for that day.</li>
</ul></li>
</ul>
<p>The primary communication for the class will go through Courseplus. That is where we will post course announcements, answer common questions, and as the primary means of communication between course participants and course instructors.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>If you are registered for the course, you should have access to Courseplus now. Once you have access you will also be able to find all material and dates/times of drop-in hours. Any Zoom links will be posted on Courseplus.</p>
</div>
</div>
<section id="course-staff" class="level3">
<h3 class="anchored" data-anchor-id="course-staff">Course Staff</h3>
<p>The course instructor this year is <a href="http://lcolladotor.github.io/">Leonardo Collado Torres</a> who is an Investigator at the <a href="https://www.libd.org/">Lieber Institute for Brain Development</a> and an Assistant Professor in the <a href="https://publichealth.jhu.edu/departments/biostatistics">Department of Biostatistics</a> at the <a href="https://publichealth.jhu.edu">Johns Hopkins Bloomberg School of Public Health</a>. I was actually a student of this course in Fall 2012 when <a href="https://bsky.app/profile/rdpeng.org">Roger D. Peng</a> was the instructor of this course and <a href="https://bsky.app/profile/kralljr.bsky.social">Jenna Krall</a> was the only teaching assistant. I first taught this course in 2023. I myself started learning and <a href="http://lcolladotor.github.io/#teaching">teaching</a> about R in 2008 and we have a lot of material in the <a href="https://research.libd.org/rstatsclub/">LIBD rstats club</a> that you might be interested in.</p>
<p>At the Lieber Institute for Brain Development (<a href="https://www.libd.org/">LIBD</a>), my group works on understanding the roots and signatures of disease (particularly psychiatric disorders) by zooming in across dimensions of gene activity. We achieve this by studying gene expression at all expression feature levels (genes, exons, exon-exon junctions, and un-annotated regions) and by using different gene expression measurement technologies (bulk RNA-seq, single cell/nucleus RNA-seq, and spatial transcriptomics) that provide finer biological resolution and localization of gene expression. We work closely with collaborators from LIBD as well as from Johns Hopkins University (JHU) and other institutions, which reflects the cross-disciplinary approach and diversity in expertise needed to further advance our understanding of high throughput biology.</p>
<p>Most weeks I use <a href="http://cran.r-project.org/">R</a> and <a href="http://www.bioconductor.org/">Bioconductor</a>, and on some days I <a href="https://lcolladotor.github.io/pkgs/">write R packages</a>. Occasionally I write <a href="http://lcolladotor.github.io/#blog">blog posts</a> about them and other tools. I’m a co-founder of the <a href="http://LieberInstitute.github.io/rstatsclub/">LIBD rstats club</a> and the <a href="https://comunidadbioinfo.github.io">CDSB community</a> of R and Bioconductor developers in Mexico and Latin America, that we described at the <a href="https://www.r-consortium.org/blog/2020/03/18/cdsb-diversity-and-outreach-hotspot-in-mexico">R Consortium website</a>. In the past, I also served on the <a href="http://bioconductor.org/about/community-advisory-board/">Bioconductor Community Advisory Board</a>.</p>
<p>If you want, you can find me on <a href="https://bsky.app/profile/lcolladotor.bsky.social">Bluesky</a>, <a href="https://www.linkedin.com/in/lcollado/">LinkedIn</a>, among other places.</p>
<section id="teaching-assistants" class="level4">
<h4 class="anchored" data-anchor-id="teaching-assistants">Teaching Assistants</h4>
<p>We also have three amazing TAs this year:</p>
<ul>
<li>Yanting Hu (<a href="mailto:yhu157@jh.edu">yhu157@jh.edu</a>): I am a second-year Master’s student in Biostatistics. My research interests focus on genomics, particularly on developing and applying statistical methods to analyze high-dimensional genomic data. In my leisure time, I enjoy photography, especially capturing everyday life.</li>
<li>Phyllis Wei (<a href="mailto:ywei43@jhu.edu">ywei43@jhu.edu</a>). She is a fifth year Ph.D.&nbsp;student in Biostatistics. She develops methods to help understand the genetic basis of complex traits and diseases and enhance disease risk prediction models through data integration. Outside of biostatistics, she enjoys hiking, baking, and visiting museums.</li>
<li>Yu Lu (<a href="mailto:ylu136@jh.edu">ylu136@jh.edu</a>): I am a fourth year Ph.D.&nbsp;student in Biostatistics. My work is focused on developing functional data analysis methods inspired by physical activity data. In my free time, I enjoy the outdoor but also cuddling with my cat, Finland. (Didn’t manage to make him wear a harness and take a walk outside yet)</li>
</ul>
</section>
</section>
<section id="assignment-due-dates" class="level3">
<h3 class="anchored" data-anchor-id="assignment-due-dates">Assignment Due Dates</h3>
<p>All course assignment due dates appear on the <strong>Schedule</strong> and <strong>Syllabus</strong>.</p>
</section>
<section id="grading" class="level3">
<h3 class="anchored" data-anchor-id="grading">Grading</h3>
<section id="philosophy" class="level4">
<h4 class="anchored" data-anchor-id="philosophy">Philosophy</h4>
<p>We believe the purpose of graduate education is to train you to be able to think for yourself and initiate and complete your own projects. We are super excited to talk to you about ideas, work out solutions with you, and help you to figure out how to produce professional data analyses. We do not think that graduate school grades are important for this purpose. This means that we do not care very much about graduate student grades.</p>
<p>That being said, we have to give you a grade so they will be:</p>
<ul>
<li>A - Excellent - 90%+</li>
<li>B - Passing - 80%+</li>
<li>C - Needs improvement - 70%+</li>
</ul>
<p>We rarely give out grades below a C and if you consistently submit work, and do your best you are very likely to get an A or a B in the course.</p>
<p>When I was a JHBSPH student in 2011-2016, I had a scholarship from my country 🇲🇽 which had specific grade requirements, so I recognize that while most employers won’t care about your grades over your grad school projects, you might have strong reasons for aiming for high grades.</p>
</section>
<section id="relative-weights" class="level4">
<h4 class="anchored" data-anchor-id="relative-weights">Relative weights</h4>
<p>The grades are based on three projects (plus one entirely optional project to help you get set up). The breakdown of grading will be</p>
<ul>
<li>33% for Project 1</li>
<li>33% for Project 2</li>
<li>34% for Project 3</li>
</ul>
<p>If you submit an project solution, it is your own work, and it meets a basic level of completeness and effort you will get 100% for that project. If you submit a project solution, but it doesn’t meet basic completeness and effort you will receive 50%. If you do not submit an solution you will receive 0%.</p>
</section>
</section>
<section id="submitting-assignments" class="level3">
<h3 class="anchored" data-anchor-id="submitting-assignments">Submitting assignments</h3>
<p>Please write up your project solutions using R Markdown. In some cases, you will compile a R Markdown file into an HTML file and submit your HTML file to the dropbox on Courseplus. In other cases, you may create an R package or website. In all of the above, when applicable, show all your code and provide as much explanation / documentation as you can.</p>
<p>For each project, we will provide a time when we download the materials. We will assume whatever version we download at that time is what you are turning in.</p>
</section>
<section id="reproducibility" class="level3">
<h3 class="anchored" data-anchor-id="reproducibility">Reproducibility</h3>
<p>We will talk about reproducibility a bit during class, and it will be a part of the homework assignments as well. Reproducibility of scientific code is very challenging, so the faculty and TAs completely understand difficulties that arise. But we think that it is important that you practice reproducible research. In particular, your project assignments should perform the tasks that you are asked to do and create the figures and tables you are asked to make as a part of the compilation of your document. We will have some pointers for some issues that have come up as we announce the projects.</p>
</section>
<section id="code-of-conduct" class="level3">
<h3 class="anchored" data-anchor-id="code-of-conduct">Code of Conduct</h3>
<p>We are committed to providing a welcoming, inclusive, and harassment-free experience for everyone, regardless of gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, ethnicity, religion (or lack thereof), political beliefs/leanings, or technology choices. We do not tolerate harassment of course participants in any form. Sexual language and imagery is not appropriate for any work event, including group meetings, conferences, talks, parties, Twitter, and other online media. This code of conduct applies to all course participants, including instructors and TAs, and applies to all modes of interaction, both in-person and online, including GitHub project repos, Slack channels, and Twitter.</p>
<p>I was also part of the Bioconductor Code of Conduct committee for a few years. You might find the Bioconductor Code of Conduct useful as it is translated into different languages by native speakers of said languages: <a href="https://bioconductor.github.io/bioc_coc_multilingual/" class="uri">https://bioconductor.github.io/bioc_coc_multilingual/</a>.</p>
<p>Course participants violating these rules will be referred to leadership of the Department of Biostatistics and the Title IX coordinator at JHU and may face expulsion from the class.</p>
<p>All class participants agree to:</p>
<ul>
<li><strong>Be considerate</strong> in speech and actions, and actively seek to acknowledge and respect the boundaries of other members.</li>
<li><strong>Be respectful</strong>. Disagreements happen, but do not require poor behavior or poor manners. Frustration is inevitable, but it should never turn into a personal attack. A community where people feel uncomfortable or threatened is not a productive one. Course participants should be respectful both of the other course participants and those outside the course.</li>
<li><strong>Refrain from demeaning, discriminatory, or harassing behavior and speech</strong>. Harassment includes, but is not limited to: deliberate intimidation; stalking; unwanted photography or recording; sustained or willful disruption of talks or other events; inappropriate physical contact; use of sexual or discriminatory imagery, comments, or jokes; and unwelcome sexual attention. If you feel that someone has harassed you or otherwise treated you inappropriately, please alert Leonardo Collado Torres.</li>
<li><strong>Take care of each other</strong>. Refrain from advocating for, or encouraging, any of the above behavior. And, if someone asks you to stop, then stop. Alert Leonardo Collado Torres if you notice a dangerous situation, someone in distress, or violations of this code of conduct, even if they seem inconsequential.</li>
</ul>
</section>
<section id="need-help" class="level3">
<h3 class="anchored" data-anchor-id="need-help">Need Help?</h3>
<p>Please speak with Leonardo Collado Torres or one of the TAs. You can also reach out to Elizabeth Stuart, chair of the department of Biostatistics or Margaret Taub, Ombudsman for the Department of Biostatistics.</p>
<p>You may also reach out to any Hopkins resource for sexual harassment, discrimination, or misconduct:</p>
<ul>
<li>JHU Sexual Assault Helpline, 410-516-7333 (confidential)</li>
<li><a href="http://sexualassault.jhu.edu/?utm_source=JHU+Broadcast+Messages+-+Synced+List&amp;utm_campaign=c9030551f7-EMAIL_CAMPAIGN_2017_12_11&amp;utm_medium=email&amp;utm_term=0_af6859b027-c9030551f7-69248741">University Sexual Assault Response and Prevention website</a></li>
<li><a href="https://johnshopkinsspeak2us.tnwreports.com/?utm_source=JHU+Broadcast+Messages+-+Synced+List&amp;utm_campaign=c9030551f7-EMAIL_CAMPAIGN_2017_12_11&amp;utm_medium=email&amp;utm_term=0_af6859b027-c9030551f7-69248741">Johns Hopkins Compliance Hotline</a>, 844-SPEAK2US (844-733-2528)</li>
<li><a href="https://jhu.us5.list-manage.com/track/click?u=bd75ef1a5cad0cbfd522412c4&amp;id=8a667a12dd&amp;e=b1124f7c17">Hopkins Policies Online</a></li>
<li><a href="https://jhu.us5.list-manage.com/track/click?u=bd75ef1a5cad0cbfd522412c4&amp;id=928bcfb8a9&amp;e=b1124f7c17">JHU Office of Institutional Equity</a> 410-516-8075 (nonconfidential)</li>
<li><a href="https://jhu.us5.list-manage.com/track/click?u=bd75ef1a5cad0cbfd522412c4&amp;id=98f4091f97&amp;e=b1124f7c17">Johns Hopkins Student Assistance Program</a> (JHSAP), 443-287-7000</li>
<li><a href="https://jhu.us5.list-manage.com/track/click?u=bd75ef1a5cad0cbfd522412c4&amp;id=d51077694c&amp;e=b1124f7c17">University Health Services</a>, 410-955-1892</li>
<li><a href="https://jhu.us5.list-manage.com/track/click?u=bd75ef1a5cad0cbfd522412c4&amp;id=af1f20bd97&amp;e=b1124f7c17">The Faculty and Staff Assistance Program</a> (FASAP), 443-997-7000</li>
</ul>
</section>
<section id="feedback" class="level3">
<h3 class="anchored" data-anchor-id="feedback">Feedback</h3>
<p>We welcome feedback on this Code of Conduct.</p>
</section>
<section id="license-and-attribution" class="level3">
<h3 class="anchored" data-anchor-id="license-and-attribution">License and attribution</h3>
<p>This Code of Conduct is distributed under a Attribution-NonCommercial-ShareAlike 4.0 International (CC BY-NC-SA 4.0) license. Portions of above text comprised of language from the Codes of Conduct adopted by rOpenSci and Django, which are licensed by CC BY-SA 4.0 and CC BY 3.0. This work was further inspired by Ada Initiative’s ‘’how to design a code of conduct for your community’’ and Geek Feminism’s Code of conduct evaluations and expanded by Ashley Johnson and Shannon Ellis in the Jeff Leek group.</p>
</section>
<section id="academic-ethics" class="level3">
<h3 class="anchored" data-anchor-id="academic-ethics">Academic Ethics</h3>
<p>Students enrolled in the Bloomberg School of Public Health of The Johns Hopkins University assume an obligation to conduct themselves in a manner appropriate to the University’s mission as an institution of higher education. A student is obligated to refrain from acts which he or she knows, or under the circumstances has reason to know, impair the academic integrity of the University. Violations of academic integrity include, but are not limited to: cheating; plagiarism; knowingly furnishing false information to any agent of the University for inclusion in the academic record; violation of the rights and welfare of animal or human subjects in research; and misconduct as a member of either School or University committees or recognized groups or organizations.</p>
<p>Students should be familiar with the policies and procedures specified under Policy and Procedure Manual Student-01 (Academic Ethics), available on the school’s <a href="https://my.jhsph.edu/Pages/Faculty.aspx">portal</a>.</p>
<p>The faculty, staff and students of the Bloomberg School of Public Health and the Johns Hopkins University have the shared responsibility to conduct themselves in a manner that upholds the law and respects the rights of others. Students enrolled in the School are subject to the Student Conduct Code (detailed in Policy and Procedure Manual Student-06) and assume an obligation to conduct themselves in a manner which upholds the law and respects the rights of others. They are responsible for maintaining the academic integrity of the institution and for preserving an environment conducive to the safe pursuit of the School’s educational, research, and professional practice missions.</p>
</section>
<section id="disability-support-service" class="level2">
<h2 class="anchored" data-anchor-id="disability-support-service">Disability Support Service</h2>
<p>Students requiring accommodations for disabilities should register with <a href="https://publichealth.jhu.edu/about/inclusion-diversity-anti-racism-and-equity-idare/student-disability-services">Student Disability Service (SDS)</a>. It is the responsibility of the student to register for accommodations with SDS. Accommodations take effect upon approval and apply to the remainder of the time for which a student is registered and enrolled at the Bloomberg School of Public Health. Once you are f a student in your class has approved accommodations you will receive formal notification and the student will be encouraged to reach out. If you have questions about requesting accommodations, please contact <code>BSPH.dss@jhu.edu</code>.</p>
</section>
<section id="previous-versions-of-the-class" class="level2">
<h2 class="anchored" data-anchor-id="previous-versions-of-the-class">Previous versions of the class</h2>
<ul>
<li><a href="https://www.stephaniehicks.com/jhustatcomputing2022" class="uri">https://www.stephaniehicks.com/jhustatcomputing2022</a></li>
<li><a href="https://www.stephaniehicks.com/jhustatcomputing2021" class="uri">https://www.stephaniehicks.com/jhustatcomputing2021</a></li>
<li><a href="https://rdpeng.github.io/Biostat776" class="uri">https://rdpeng.github.io/Biostat776</a></li>
</ul>
</section>
<section id="typos-and-corrections" class="level2">
<h2 class="anchored" data-anchor-id="typos-and-corrections">Typos and corrections</h2>
<p>Feel free to submit typos/errors/etc via the GitHub repository associated with the class: <a href="https://github.com/lcolladotor/jhustatcomputing/issues" class="uri">https://github.com/lcolladotor/jhustatcomputing/issues</a>. You will have the thanks of your grateful instructor!</p>
</section>
</section>
<section id="r-session-information" class="level1">
<h1>R session information</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">width =</span> <span class="dv">120</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>sessioninfo<span class="sc">::</span><span class="fu">session_info</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.5.1 (2025-06-13)
 os       macOS Sequoia 15.6.1
 system   aarch64, darwin20
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/New_York
 date     2025-08-26
 pandoc   3.4 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)
 quarto   1.4.550 @ /usr/local/bin/quarto

─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version date (UTC) lib source
 cli           3.6.5   2025-04-23 [1] CRAN (R 4.5.0)
 colorout    * 1.3-2   2025-05-09 [1] Github (jalvesaq/colorout@572ab10)
 digest        0.6.37  2024-08-19 [1] CRAN (R 4.5.0)
 evaluate      1.0.4   2025-06-18 [1] CRAN (R 4.5.0)
 fastmap       1.2.0   2024-05-15 [1] CRAN (R 4.5.0)
 htmltools     0.5.8.1 2024-04-04 [1] CRAN (R 4.5.0)
 htmlwidgets   1.6.4   2023-12-06 [1] CRAN (R 4.5.0)
 jsonlite      2.0.0   2025-03-27 [1] CRAN (R 4.5.0)
 knitr         1.50    2025-03-16 [1] CRAN (R 4.5.0)
 rlang         1.1.6   2025-04-11 [1] CRAN (R 4.5.0)
 rmarkdown     2.29    2024-11-04 [1] CRAN (R 4.5.0)
 rstudioapi    0.17.1  2024-10-22 [1] CRAN (R 4.5.0)
 sessioninfo   1.2.3   2025-02-05 [1] CRAN (R 4.5.0)
 xfun          0.52    2025-04-02 [1] CRAN (R 4.5.0)
 yaml          2.3.10  2024-07-26 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/library
 * ── Packages attached to the search path.

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/lcolladotor\.github\.io\/jhustatcomputing");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>